### PR TITLE
Create + Delete Workspace

### PR DESCRIPTION
**Pull request by jate-koh**
### What
Briefly describe this feature / fix / refactor

- In brief, this branch at API endpoints for creating and deleting workspaces for admins and owners.

### Why
Describe the reason why we need this feature / fix / refactor

- These features are included in stakeholders' defined requirements. Thus, it is a vital feature of Codern.

### How
Describe the feature / fix / refactor in detail.

- Explain the technical solution you have done to create / fix this pull request
   - In the df6bdd3, the `CREATE` endpoint are added. Using POST method to create a new workspace. As of now, noted that **anyone can create workspace**, since there're no function for checking permission in `Create()`.
   - In commit 1c0080f and 1a9a9fc, are fixes for the already implemented `UPDATE` API route. The codes are not in sync to the newer commits (esp. new `RawWorkspace` Struct in `domain`) so I took the opportunity to refactor and fix it.
   - 682f81e and ea5b735 are commits that involved adding `DELETE` **(soft delete)** route. This includes adding new column called `is_delete` to the database (can viewed in `db/migrations` folder).
   - In bcfcbe8, I added `CheckPermRole()` just to checking user's perms; however, it's different from `CheckPerm()`. In its parameter we can specified what role we want to check for. For example, if I want to check for just `OWNER` individually:
```go
isAuthorized, err := u.CheckPermRole(userId, workspaceId, []domain.WorkspaceRole{domain.OwnerRole})
```

- Attach the screenshot (if applicable)
   - If requested, I'll provide it in the comment sections 👍🏻

### Todo list (if applicable)
- [x] Open my computer
- [x] Add `CREATE` workspace API route
- [x] Add `DELETE` workspace API route
- [x] Add perms checking based on specified roles in parameters. 

## Checklist
- [x] Drink juicy beverage when coding ☕️
- [x] Explained the purpose of this PR
- [x] Tested on **Local machine** and verified that there're no visible errors
- [ ] Tested on **Staging server** and verified that there're no visible errors
- [ ] Updated documentation (if applicable)
- [ ] Added or updated test suite for the changes (if applicable)

### CODE REVIEW

File: internal/constant/constant.go [hunk 1]
```diff
@@ -18,4 +18,6 @@ var (
 	SeaweedFsChunkSize      = 1048576 // 1 MiB
 
 	MaxInvitationCodeChar = 6
+
+	DefaultProfileUrl = "/workspaces/1/profile"
```
Comment (riflowth) [inline internal/constant/constant.go:22]: Can we use `/workspaces/default/profile` instead ?

File: domain/workspace.go [hunk 2]
```diff
@@ -26,8 +28,14 @@ type Workspace struct {
 	RecentlyVisitedAt   time.Time `json:"recentlyVisitedAt" db:"recently_visited_at"`
 }
 
+type CreateWorkspace struct {
+	Name    string `validate:"required"`
```
Comment (riflowth) [inline domain/workspace.go]: Validate tag should be in payload only, please remove it

File: internal/constant/constant.go (hunk 1 above)
Comment (riflowth) [inline internal/constant/constant.go:22]: Can we use `/workspaces/default/profile` instead?

File: repository/workspace.go [hunk 3]
```diff
@@ -17,6 +17,26 @@ func NewWorkspaceRepository(db *sqlx.DB) domain.WorkspaceRepository {
...
+	_, err = r.db.Exec(`
+		INSERT INTO workspace_participant (workspace_id, user_id, role, favorite, joined_at, recently_visited_at)
+		VALUES (?, ?, ?, ?, ?, ?)
+	`, workspace.Id, userId, domain.OwnerRole, false, time.Now(), time.Now())
+	if err != nil {
+		return fmt.Errorf("cannot query to insert owner into created workspace: %w", err)
+	}
+
+	return nil
+}
```
Comment (riflowth) [inline repository/workspace.go]: Please wrap this operation with transaction

**Review by riflowth — APPROVED** [review]
GOOD IMPLEMENTATION, LGTM!
We need this feature lol

**Comment by riflowth** [comment]
Ops, it should be `request changes` not `approve` my bad @jate-koh

File: internal/constant/constant.go (hunk 1 above)
Comment (jate-koh) [reply internal/constant/constant.go:22]: I am afraid `fiber`'s `ParamsParser` doesn't allow that. Is there any workaround? 

Can we perhaps make a route for specifically getting default assets/files /pictures? Isn't that a tad bit too unnecessary?

Screenshot 2567-02-21 at 09 34 33

File: domain/workspace.go (hunk 2 above)
Comment (jate-koh) [reply domain/workspace.go]: Done. Removed validate tag in 6b7dc40

File: repository/workspace.go (hunk 3 above)
Comment (jate-koh) [reply repository/workspace.go]: Done. Resolved in commit 98c411b.

File: internal/constant/constant.go (hunk 1 above)
Comment (jate-koh) [reply internal/constant/constant.go:22]: I tried using id 0 as you requested too. But to no avail. It seem like `validator.Struct` in `internal/validator/payload.go` wouldn't count payload with id 0 as `valid payload`

Comment (riflowth) [reply internal/constant/constant.go:22]: Ok, so ID 1 as the default workspace id is acceptable.

**Pull request merged**